### PR TITLE
Update Namespace names to follow DNS label format

### PR DESCRIPTION
**Pull request by vishh**
Related issues: #8010 #6666 

cc @bgrant0607 @thockin

### CODE REVIEW

**Comment by vishh** [comment]
cc @ArtfulCoder

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func ValidateNodeName(name string, prefix bool) (bool, string) {
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateNamespaceName(name string, prefix bool) (bool, string) {
-	return nameIsDNSSubdomain(name, prefix)
+	return nameIsDNS952Label(name, prefix)
```
Comment (thockin) [inline pkg/api/validation/validation.go]: Should be IsDNS1123Label() I think - no reason for the more restrictive format that I can see.

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -212,7 +212,7 @@ func ValidateObjectMeta(meta *api.ObjectMeta, requiresNamespace bool, nameFn Val
 	if requiresNamespace {
 		if len(meta.Namespace) == 0 {
 			allErrs = append(allErrs, errs.NewFieldRequired("namespace"))
-		} else if !util.IsDNS1123Subdomain(meta.Namespace) {
+		} else if !util.IsDNS952Label(meta.Namespace) {
```
Comment (thockin) [inline pkg/api/validation/validation.go]: Can we thunk this through to ValidateNamespaceName() ?

File: pkg/api/v1beta3/types.go [hunk 3]
```diff
@@ -99,7 +99,7 @@ type ObjectMeta struct {
 	// equivalent to the "default" namespace, but "default" is the canonical representation.
 	// Not all objects are required to be scoped to a namespace - the value of this field for
 	// those objects will be empty.
-	Namespace string `json:"namespace,omitempty" description:"namespace of the object; cannot be updated"`
+	Namespace string `json:"namespace,omitempty" description:"namespace of the object; must be a DNS_LABEL; 'default' by default; cannot be updated"`
```
Comment (bgrant0607) [inline pkg/api/v1beta3/types.go]: There is no default namespace in v1beta3. It must be explicit.

File: pkg/api/v1/types.go [hunk 4]
```diff
@@ -99,7 +99,7 @@ type ObjectMeta struct {
 	// equivalent to the "default" namespace, but "default" is the canonical representation.
 	// Not all objects are required to be scoped to a namespace - the value of this field for
 	// those objects will be empty.
-	Namespace string `json:"namespace,omitempty" description:"namespace of the object; cannot be updated"`
+	Namespace string `json:"namespace,omitempty" description:"namespace of the object; must be a DNS_LABEL; 'default' by default; cannot be updated"`
```
Comment (bgrant0607) [inline pkg/api/v1/types.go]: As with v1beta3, there is no default.

Comment (vishh) [inline pkg/api/v1/types.go]: Ack. updated

File: pkg/api/v1beta3/types.go (hunk 3 above)
Comment (vishh) [inline pkg/api/v1beta3/types.go]: Ack. updated

File: pkg/api/validation/validation.go (hunk 2 above)
Comment (vishh) [inline pkg/api/validation/validation.go]: Done

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (vishh) [inline pkg/api/validation/validation.go]: Done

**Comment by vishh** [comment]
Comments addressed. PTAL @thockin

File: pkg/api/validation/validation.go [hunk 5]
```diff
@@ -112,7 +112,7 @@ func ValidateReplicationControllerName(name string, prefix bool) (bool, string)
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateServiceName(name string, prefix bool) (bool, string) {
-	return nameIsDNS952Label(name, prefix)
+	return nameIsDNS1123Label(name, prefix)
```
Comment (thockin) [inline pkg/api/validation/validation.go]: Sorry, services are properly DNS952 for now.  That's on purpose (1123 allows leading digits which we don't handle)

**Comment by thockin** [comment]
ping

File: pkg/api/validation/validation.go (hunk 5 above)
Comment (vishh) [inline pkg/api/validation/validation.go]: Done

**Comment by vishh** [comment]
@thockin: Code updated.

File: pkg/api/validation/validation_test.go [hunk 6]
```diff
@@ -1426,7 +1455,7 @@ func TestValidateService(t *testing.T) {
 		{
 			name: "too long name",
 			tweakSvc: func(s *api.Service) {
-				s.Name = strings.Repeat("a", 25)
+				s.Name = strings.Repeat("a", 65)
```
Comment (thockin) [inline pkg/api/validation/validation_test.go]: you should not need to change this

File: pkg/api/validation/validation_test.go [hunk 7]
```diff
@@ -1440,7 +1469,7 @@ func TestValidateService(t *testing.T) {
 		{
 			name: "too long generateName",
 			tweakSvc: func(s *api.Service) {
-				s.GenerateName = strings.Repeat("a", 25)
+				s.GenerateName = strings.Repeat("a", 65)
```
Comment (thockin) [inline pkg/api/validation/validation_test.go]: or this

File: pkg/api/validation/validation.go [hunk 8]
```diff
@@ -176,6 +176,17 @@ func nameIsDNSSubdomain(name string, prefix bool) (bool, string) {
 	return false, dnsSubdomainErrorMsg
 }
 
+// nameIsDNS1123Label is a ValidateNameFunc for names that must be a DNS 1123 label.
+func nameIsDNS1123Label(name string, prefix bool) (bool, string) {
```
Comment (thockin) [inline pkg/api/validation/validation.go]: nit: nameIsDNSLabel to match above

**Comment by thockin** [comment]
a couple nits

File: pkg/api/validation/validation_test.go (hunk 6 above)
Comment (vishh) [inline pkg/api/validation/validation_test.go]: Done

File: pkg/api/validation/validation_test.go (hunk 7 above)
Comment (vishh) [inline pkg/api/validation/validation_test.go]: Done

**Comment by vishh** [comment]
Fixed.

File: pkg/api/validation/validation_test.go (hunk 7 above)
Comment (thockin) [inline pkg/api/validation/validation_test.go]: Not pushed yet?

File: pkg/api/validation/validation.go (hunk 8 above)
Comment (thockin) [inline pkg/api/validation/validation.go]: I'll let this inconsistency slide.  I'm ambivalent about which is better, but it's inconsistent as-is.

nameIsDNSSubdomain vs nameIsDNS1123Label

**Comment by thockin** [comment]
LGTM, merge on green.  Self-merge if I forget.

File: pkg/api/validation/validation.go (hunk 8 above)
Comment (vishh) [inline pkg/api/validation/validation.go]: Updated to `nameIsDNSLabel`.

**Pull request merged**